### PR TITLE
Add API changelog for streamly-0.10.0 and streamly-core-0.2.0

### PR DESCRIPTION
**Pull request by harendra-kumar**

### CODE REVIEW

File: core/docs/ApiChangelogs/0.1.0-0.2.0.txt [hunk 1]
```diff
@@ -0,0 +1,1471 @@
+---------------------------------
+API Annotations
+---------------------------------
+
+[A] : Added
+[R] : Removed
+[C] : Signature changed
```
Comment (adithyaov) [inline core/docs/ApiChangelogs/0.1.0-0.2.0.txt:7]: Just "Changed" should work. The tag is present on modules as well. This is basically a tree and the node has `[C]` if any descendants of it are changed.

File: core/docs/ApiChangelogs/0.1.0-0.2.0.txt [hunk 2]
```diff
@@ -0,0 +1,1471 @@
...
+[O] : Old signature
+[N] : New signature
```
Comment (adithyaov) [inline core/docs/ApiChangelogs/0.1.0-0.2.0.txt:9]: `signature` is very specific to the context.

File: core/docs/ApiChangelogs/0.1.0-0.2.0.txt [hunk 3]
```diff
@@ -0,0 +1,1471 @@
...
+[C] Streamly.Data.Stream
+    [A] wordsBy :: Monad m => (a -> Bool) -> Fold m a b -> Stream m a -> Stream m b
+    [A] splitOn :: Monad m => (a -> Bool) -> Fold m a b -> Stream m a -> Stream m b
+    [C] handle
+        [O] handle :: (MonadCatch m, Exception e) => (e -> Stream m a) -> Stream m a -> Stream m a
+        [N] handle :: (MonadCatch m, Exception e) => (e -> m (Stream m a)) -> Stream m a -> Stream m a
+    [A] groupsOf :: Monad m => Int -> Fold m a b -> Stream m a -> Stream m b
+[C] Streamly.Data.ParserK
+    [D] fromParser :: (MonadIO m, Unbox a) => Parser a m b -> ParserK (Array a) m b
+    [D] fromFold :: (MonadIO m, Unbox a) => Fold m a b -> ParserK (Array a) m b
```
Comment (adithyaov) [inline core/docs/ApiChangelogs/0.1.0-0.2.0.txt:56]: If we deprecate `fromFold` we should have corresponding required parser combinators. Like, `drain`, `all`, `any` etc.
Why can't folds be converted to parsers, again?
`Fold initial step extract final` = `Parser initial step final`
Does the Fold applicative make sense anymore, given scanning is what we have in mind?

File: docs/User/Project/ApiChangelogs/0.9.0-0.10.0.txt [hunk 4]
```diff
@@ -0,0 +1,424 @@
+---------------------------------
+API Annotations
+---------------------------------
+
+[A] : Added
+[R] : Removed
+[C] : Signature changed
+[O] : Old signature
+[N] : New signature
```
Comment (adithyaov) [inline docs/User/Project/ApiChangelogs/0.9.0-0.10.0.txt:9]: See comments above.

File: core/docs/ApiChangelogs/0.1.0-0.2.0.txt (hunk 3 above)
Comment (harendra-kumar) [reply core/docs/ApiChangelogs/0.1.0-0.2.0.txt:56]: instead of K.fromFold we use D.fromFold and `K.adapt` now.

**Pull request merged**